packages/dappmanager/src/modules/docker/utils.ts

```diff
@@ -68,3 +71,59 @@ export function getDockerTimeoutMax(
   }
   return timeout;
 }
+
+/**
+ * Returns the ports value returned by the docker api for a specific container.
+ * Example of ports: Dockerode.Ports[]
+ *   - "IP": "0.0.0.0" => for ipv4
+ *   - "IP": "::" => for ipv6
+ * ```
+ * "Ports": [
+      {
+        "PrivatePort": 80,
+        "Type": "tcp"
+      },
+      {
+        "IP": "0.0.0.0",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      },
+      {
+        "IP": "::",
+        "PrivatePort": 8000,
+        "PublicPort": 2345,
+        "Type": "tcp"
+      }
+    ]
+ * ```
+ */
+export function parseDockerApiListPorts(
```

Comment:
This does not de-duplicate ports. You must ensure that the UI condition of one port per protocol and container port.

Reply:
right! done